tests/sentry/middleware/test_customer_domain.py

```diff
@@ -22,6 +24,30 @@ def test_sets_active_organization_if_exists(self):
         assert request.session == {"activeorg": "test"}
         assert response == request
 
+    def test_recycles_last_active_org(self):
+        self.create_organization(name="test")
+
+        request = RequestFactory().get("/organizations/test/issues/")
+        request.subdomain = "does-not-exist"
+        request.session = {"activeorg": "test"}
+        response = CustomerDomainMiddleware(lambda request: request)(request)
+
+        assert request.session == {"activeorg": "test"}
+        assert response.status_code == 302
+        assert response["Location"] == "http://test.testserver/organizations/test/issues/"
+
+    def test_recycles_last_active_org_path_mismatch(self):
+        self.create_organization(name="test")
+
+        request = RequestFactory().get("/organizations/albertos-apples/issues/")
+        request.subdomain = "does-not-exist"
+        request.session = {"activeorg": "test"}
+        response = CustomerDomainMiddleware(lambda request: request)(request)
```

Comment:
Could overwriting the subdomain and path segment with the current session value impact how switching organizations works?

Reply:
@markstory the `activeorg` session value is only used as a fallback. The `_resolve_activeorg()` function encapsulates which org should be resolved.